src/Compiler/test/Microsoft.AspNetCore.Razor.Test.Common/Language/IntegrationTests/RazorIntegrationTestBase.cs

```diff
@@ -419,7 +419,7 @@ public override string Message
                 var builder = new StringBuilder();
                 builder.AppendLine("Compilation failed: ");
 
-                var diagnostics = Compilation.GetDiagnostics().Concat(Diagnostics ?? Enumerable.Empty<Diagnostic>());
+                var diagnostics = Compilation.GetDiagnostics().Concat(Diagnostics);
```

Comment:
Does `Concat` handle `default(ImmutableArray<>)` correctly? Or does it throw?

Reply:
Good catch, it throws. But `ImmutableArray.Empty` cannot be used as default parameter, but I can replace `default` by `Empty` in the constructor: https://github.com/dotnet/razor-tooling/pull/6814/commits/b771221d81b098f4138f4b16ca827342028caaf8 (seems there is no `EmptyIfDefault` helper method for this case?)

Reply:
We have a `NullToEmpty()` extension in Roslyn, but it doesn't appear there's one here.

Reply:
I would vote to add one.

Reply:
Should I add it as part of this PR?

Reply:
Up to you.